assets/dns/daemonset.yaml

```diff
@@ -127,7 +127,7 @@ spec:
             # Update /etc/hosts only if we get valid service IPs
             # We will not update /etc/hosts when there is coredns service outage or api unavailability
             # Stale entries could exist in /etc/hosts if the service is deleted
-            if [[ "${#svc_ips[@]}" -ne 0 ]]; then
+            if [[ -n "${svc_ips[*]}" ]]; then
```

Comment:
Why is this necessary?  This works for me on RHEL76:
```bash
$ echo $BASH_VERSION
4.2.46(2)-release
$ set -u
$ declare -A svc_ips
$ [[ "${#svc_ips[@]}" -ne 0 ]]
$ echo $?
1
$
```
Do we need to report a BZ against RHEL8's Bash?

Reply:
much newer bash, probably expected

Reply:
I filed [BZ#1875566](https://bugzilla.redhat.com/show_bug.cgi?id=1875566), "Bash says a declared but empty array is unbound".

Reply:
The behavior in 4.2 was a bug. It was fixed incorrectly in 4.3 (making both `${#svc_ips[@]}` and `${svc_ips[@]}` an error) and then fully fixed in 4.4 (making the former an error but the latter not). This has to do with POSIX compliance or something.

Reply:
This PR seems to have fixed clusters using the RHEL 8 base, but broken clusters using the prior RHEL 7 base:
eg
```
sh-4.2# echo $BASH_VERSION
4.2.46(2)-release
sh-4.2# set -u
sh-4.2# declare -A svc_ips
sh-4.2# [[-n "${svc_ips[*]}"]]
sh: svc_ips[*]: unbound variable
```

Reply:
Is there any reason for something to be building images with UBI7 base and cluster-dns-operator git master?

Not sure if there's any alternative that would work in both versions. Simplest fix, if needed, would probably be to just set an additional variable in the earlier loop to indicate "I added at least one IP to `svc_ips`", and then test that here instead of testing `svc_ips` directly.

Reply:
Looks like 4.6 nightly builds are still using UBI7, but that should to change to UBI8 this week to match CI, so we should actually be able to leave this fix as is. 

https://coreos.slack.com/archives/CCH60A77E/p1599489624207700 for context .

Reply:
Would `[[ -n "${svc_ips[*]-}" ]]` work on both Bash 4.2 and 4.4?

Reply:
> Would `[[ -n "${svc_ips[*]-}" ]]` work on both Bash 4.2 and 4.4?

I think so... what exactly does the `-` change?

Reply:
`${parameter-default}` means to substitute the value of the variable `parameter` if it is set, or the literal value `default` (which in our case is the empty string) if `parameter` is not set.

Reply:
Ah, I had tried changing the original `${#svc_ips[@]}` to `${#svc_ips[@]:-0}`, which does not work.

Reply:
https://github.com/openshift/cluster-dns-operator/pull/193 changes the test to `[[ -n "${svc_ips[*]-}" ]]`.